plaso/storage/redis/writer.py

```diff
@@ -0,0 +1,362 @@
+# -*- coding: utf-8 -*-
+"""Storage writer for Redis."""
+
+from __future__ import unicode_literals
+
+from plaso.lib import definitions
+from plaso.storage import interface
+from plaso.storage.redis import redis_store
+
+
+class RedisStorageWriter(interface.StorageWriter):
+  """Redis-based storage writer."""
+
+  def __init__(
+      self, session, storage_type=definitions.STORAGE_TYPE_TASK, task=None):
+    """Initializes a storage writer.
+
+    Args:
+      session (Session): session the storage changes are part of.
+      storage_type (Optional[str]): storage type.
+      task (Optional[Task]): task.
+
+    Raises:
+      RuntimeError: if no task is provided.
+    """
+    if not task:
+      raise RuntimeError('Task required.')
+
+    super(RedisStorageWriter, self).__init__(
+        session=session, storage_type=storage_type, task=task)
+
+    self._redis_namespace = '{0:s}-{1:s}'.format(
+        task.session_identifier, task.identifier)
+    self._store = None
+
+  def CreateTaskStorage(self, task, task_storage_format):
+    """Creates a task storage.
+
+    Args:
+      task (Task): task.
+      task_storage_format (str): storage format to store task results.
+
+    Raises:
+      IOError: always, as creating a task is not supported by the redis store.
+      OSError: always, as creating a task is not supported by the redis store.
+    """
+    raise IOError(
+        'Creating a task storage is not supported by the redis store.')
+
+  def FinalizeTaskStorage(self, task):
+    """Finalizes a processed task storage.
+
+    Args:
+      task (Task): task.
+
+    Raises:
+      IOError: always, as creating a finalizing a task storage is not supported
+          by the redis store.
+      OSError: always, as creating a finalizing a task storage is not supported
+          by the redis store.
+    """
+    raise IOError(
+        'Finalizing a task storage is not supported by the redis store.')
+
+  # pylint: disable=arguments-differ
+  def Open(self, redis_client=None):
+    """Opens the storage writer.
+
+    Raises:
+      IOError: if the storage writer is already opened.
+      OSError: if the storage writer is already opened.
+    """
+    if self._store:
+      raise IOError('Storage writer already opened.')
+
+    self._store = redis_store.RedisStore(
+        storage_type=self._storage_type,
+        session_identifier=self._task.session_identifier,
+        task_identifier=self._task.identifier)
+
+    self._store.Open(redis_client=redis_client)
+
+  def Close(self):
+    """Closes the storage writer.
+
+    Raises:
+      IOError: if the storage writer is closed.
+      OSError: if the storage writer is closed.
+    """
+    if not self._store:
+      raise IOError('Storage writer is not open.')
+
+    self._store.Finalize()
+    self._store = None
+
+  def PrepareMergeTaskStorage(self, task):
+    """Prepares a task storage for merging.
+
+    Args:
+      task (Task): task.
+    """
+    # No preparations are necessary.
```

Comment:
add return to indicate this is an empty method

Reply:
Done